http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/CrtRequestExecutor.java

```diff
@@ -205,55 +91,28 @@ private void executeRequest(CrtRequestContext executionContext,
 
             // Submit the request on the connection
             crtConn.makeRequest(crtRequest, crtResponseHandler).activate();
-        } catch (HttpException e) {
-            crtConn.close();
-
-            Throwable toThrow = wrapWithIoExceptionIfRetryable(e);
-            requestFuture.completeExceptionally(toThrow);
-        } catch (IllegalStateException | CrtRuntimeException e) {
-            crtConn.close();
-            requestFuture.completeExceptionally(e);
+        } catch (Throwable throwable) {
+            handleException(requestFuture, crtConn, throwable);
         }
     }
 
-    /**
-     * Create the execution future and set up the cancellation logic.
-     * @return The created execution future.
-     */
-    private CompletableFuture<Void> createAsyncExecutionFuture(AsyncExecuteRequest request) {
-        CompletableFuture<Void> future = new CompletableFuture<>();
+    private static void handleException(CompletableFuture<SdkHttpFullResponse> requestFuture, HttpClientConnection crtConn,
+                          Throwable throwable) {
 
-        future.whenComplete((r, t) -> {
-            if (t == null) {
-                return;
-            }
+        crtConn.close();
 
-            // TODO: Aborting request once it's supported in CRT
-            if (future.isCancelled()) {
-                request.responseHandler().onError(new SdkCancellationException("The request was cancelled"));
-            }
-        });
-
-        return future;
-    }
-
-    /**
-     * Notify the provided response handler and future of the failure.
-     */
-    private void reportAsyncFailure(HttpClientConnection crtConn,
-                               Throwable cause,
-                               CompletableFuture<Void> executeFuture,
-                               SdkAsyncHttpResponseHandler responseHandler) {
-        if (crtConn != null) {
-            crtConn.close();
+        if (throwable instanceof HttpException) {
+            Throwable toThrow = wrapWithIoExceptionIfRetryable((HttpException) throwable);
+            requestFuture.completeExceptionally(toThrow);
+            return;
         }
 
-        try {
-            responseHandler.onError(cause);
-        } catch (Exception e) {
-            log.error(() -> "SdkAsyncHttpResponseHandler " + responseHandler + " threw an exception in onError. It will be "
-                            + "ignored.", e);
+        if (throwable instanceof CrtRuntimeException || throwable instanceof IllegalStateException) {
+            // CRT throws IllegalStateException if the connection is closed
+            requestFuture.completeExceptionally(new IOException(throwable));
+            return;
         }
```

Comment:
This is the new code to wrap them so that they can be retried. This matches with async behavior.